pkg/api/dashboard.go

```diff
@@ -77,6 +77,18 @@ func (hs *HTTPServer) GetDashboard(c *models.ReqContext) response.Response {
 	if rsp != nil {
 		return rsp
 	}
+
+	var (
+		hasPublicDashboard bool
+		err                error
+	)
+	if hs.Features.IsEnabled(featuremgmt.FlagPublicDashboards) {
+		hasPublicDashboard, err = hs.DashboardService.HasActivePublicDashboard(c.Req.Context(), dash.Uid)
```

Comment:
Should this be in public dashboard service instead of dashboard service?

Reply:
I was considering this and I think it could go either way. Depends how you want to define your relationships. Since the relationship belongs to the dashboard resource, I kept it in the dashboard service. Happy to hear arguments in favour of putting in on the publicdashboards service.

Reply:
I have arguments for both sides, but I don't feel particularly strongly. I'm slightly leaning slightly toward public dashboard service.

Argument for public dashboard service: 
* We were told dashboard service was bloated and asked to move things over- I think this is mostly the public dashboard stuff. This is clearly related to dashboard, but it might make sense to ask the public dashboard service if a given dashboard uid has a public dashboard.
* Dashboard service doesn't know anything about dashboard_public table
* I could see as this becomes more complicated the logic behind determining whether HasActivePublicDashboard becomes more complicated

Argument for dashboard service:
* it's already there :)
* It's currently straight forward currently
* we would need to plumb the public dashboard service into the http server in order to call it in this method.

Reply:
Agreed during review. Move into public dashboard service